app/parse.py

```
@@ -15,9 +19,139 @@ class Product:
     num_of_reviews: int
 
 
-def get_all_products() -> None:
-    pass
+FIELDS = [field.name for field in fields(Product)]
+BASE_URL = "https://webscraper.io"
+HOME_URL = urljoin(BASE_URL, "/test-sites/e-commerce/more/")
+COMPUTERS_URL = urljoin(HOME_URL, "./computers/")
+PHONES_URL = urljoin(HOME_URL, "./phones/")
+LAPTOPS_URL = urljoin(COMPUTERS_URL, "./laptops/")
+TABLETS_URL = urljoin(COMPUTERS_URL, "./tablets/")
+TOUCH_URL = urljoin(PHONES_URL, "./touch/")
+
+
+def parse_page(driver: WebDriver, page_url: str) -> list[Product]:
+    wait = WebDriverWait(driver, 1)
+    driver.get(page_url)
+
+    try:
+        button = wait.until(
+            ec.element_to_be_clickable((By.CLASS_NAME, "acceptCookies"))
+        )
+        button.click()
+    except TimeoutException:
```

Waiting for and clicking the cookie consent uses `By.CLASS_NAME, "acceptCookies"`. Class names for consent modals can differ; this selector may fail silently. Consider using a more robust locator (CSS selector or XPath that matches button text), or try multiple possible selectors. Also handle StaleElementReferenceException when clicking.